packages/react-dom/src/events/EnterLeaveEventPlugin.js

```diff
@@ -169,11 +169,14 @@ const EnterLeaveEventPlugin = {
 
     accumulateEnterLeaveDispatches(leave, enter, from, to);
 
-    if (nativeEvent === lastNativeEvent) {
-      lastNativeEvent = null;
+    if (seenNativeEvents.has(nativeEvent)) {
+      seenNativeEvents.remove(nativeEvent);
       return [leave];
     }
-    lastNativeEvent = nativeEvent;
+    if (seenNativeEvents instanceof Set) {
+      seenNativeEvents.clear();
```

Comment:
This replicates the ‘overwrite’ behavior of `lastNativeEvent = nativeEvent` for clients that don't support `WeakMap`.